Source/Core/Common/HttpRequest.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <curl/curl.h>
+
+#include "Common/CommonTypes.h"
+
+namespace Common
+{
+class HttpRequest final
+{
+public:
+  HttpRequest();
+  ~HttpRequest();
+  bool IsValid() const;
+
+  using Response = std::optional<std::vector<u8>>;
+  Response Get(const std::string& url);
+  Response Post(const std::string& url, const std::vector<u8>& payload);
+  Response Post(const std::string& url, const std::string& payload);
+
+private:
+  enum class Method
+  {
+    GET,
+    POST,
+  };
+  Response Fetch(const std::string& url, Method method, const u8* payload, size_t size);
+
+  std::unique_ptr<CURL, decltype(&curl_easy_cleanup)> m_curl{curl_easy_init(), curl_easy_cleanup};
```

Comment:
You may want to consider the [PImpl](http://en.cppreference.com/w/cpp/language/pimpl) idiom here for private members/functions/etc, which would let you completely hide the fact cURL is being used. This would also remove the need to include its library header within this header.